src/gluonts/model/renewal/_predictor.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import itertools
+from functools import partial
+from pathlib import Path
+from typing import List, Optional, Callable, Iterator
+
+import mxnet as mx
+import numpy as np
+
+from gluonts.core.component import DType
+from gluonts.core.serde import load_json
+from gluonts.dataset.common import DataEntry, Dataset
+from gluonts.dataset.loader import InferenceDataLoader
+from gluonts.model.forecast import Forecast
+from gluonts.model.forecast_generator import (
+    ForecastGenerator,
+    SampleForecastGenerator,
+)
+from gluonts.mx.batchify import batchify
+from gluonts.mx.context import get_mxnet_context
+from gluonts.mx.model.predictor import RepresentableBlockPredictor
+from gluonts.transform import Transformation
+
+
+class DeepRenewalProcessSampleOutputTransform:
+    """
+    Convert a deep renewal process sample that is composed of dense interval-size
+    samples to a sparse forecast. In practice, this often only means taking the
+    first few time steps of each sample trajectory and converting them to the
+    sparse (intermittent) representation. Converts a (N, S, 2, T) array corresponding
+    to the interval-size format to a (N, S, T) array.
+    """
```

Comment:
Any way this can be described in a more general sense? Maybe other models could use this.

Also the name could be more general maybe? It's essentially the inverse transformation of the other one you had which encodes dense-but-plenty-of-zeros arrays into interval-size ones, right?

Reply:
It is the inverse transform, yes, with the caveat that it 'truncates' the 'dense' representation discarding most of the 'issue points' (points with nonzero demand). In this sense, it's specific and the only reason it's here is that there's no other nice way to make everything hybridizable imo. I highly doubt it could be used anywhere else but I can generalize it if you like.